parsers/SG.py

```diff
@@ -267,6 +232,79 @@ def fetch_price(zone_key='SG', session=None, target_datetime=None, logger=None)
     }
 
 
+def __detect_datetime_from_solar_image(solar_image, logger):
+    w, h = solar_image.size
+    crop_left = int(w * 0.75)
+    crop_top = int(h * 0.87)
+    crop_right = int(w * 0.93)
+    crop_bottom = int(h * 0.92)
+    time_img = solar_image.crop((crop_left, crop_top, crop_right, crop_bottom))
+    processed_img = __preprocess_image_for_ocr(time_img)
+    text = image_to_string(processed_img, lang='eng', config='--psm 7 -c tessedit_char_whitelist="0123456789:- "')
+
+    try:
+        time_pattern = r'\d+-\d+-\d+\s+\d+:\d+'
+        time_string = re.search(time_pattern, text, re.MULTILINE).group(0)
+    except AttributeError:
+        msg = 'Unable to get values for SG solar from OCR text: {}'.format(text)
+        logger.warning(msg, extra={'key': 'SG'})
+        return None
+
+    solar_dt = arrow.get(time_string).replace(tzinfo='Asia/Singapore')
+    return solar_dt
+
+
+def __detect_output_from_solar_image(solar_image, logger):
+    w, h = solar_image.size
+    crop_left = int(w * 0.65)
+    crop_top = int(h * 0.74)
+    crop_right = int(w * 0.93)
+    crop_bottom = int(h * 0.80)
+    output_img = solar_image.crop((crop_left, crop_top, crop_right, crop_bottom))
+    processed_img = __preprocess_image_for_ocr(output_img)
+    text = image_to_string(processed_img, lang='eng', config='--psm 7')
```

Comment:
Originally, I added the [page segmentation method](https://tesseract-ocr.github.io/tessdoc/ImproveQuality#page-segmentation-method) config to try and improve the "OMWac" issue in the middle of the night when the decimal place vanishes from the PV output.

It didn't really help much, but I've left it in, since I think `--psm 7` is most descriptive of the changes I made to cropping the output line and datetime lines separately, "Treat the image as a single text line."

I also played around with `tessedit_char_whitelist` but that did not improve the "OMWac" issue. I really wish we could use `user-patterns` to restrict character recognition to a regex-like pattern. However, [user-patterns can only be passed in via a file reference](https://tesseract-ocr.github.io/tessdoc/APIExample-user_patterns.html#cli-example). Adding that file and accounting for all the different relative paths that may appear felt like too much complexity for a scenario that's easily handled by the existing `try`/`except` block below.